components/raftstore-v2/src/worker/tablet_gc.rs

```diff
@@ -241,3 +274,67 @@ where
         Duration::from_secs(10)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use engine_test::{
+        ctor::{CfOptions, DbOptions},
+        kv::TestTabletFactory,
+    };
+    use engine_traits::{MiscExt, TabletContext, TabletRegistry};
+    use tempfile::Builder;
+
+    use super::*;
+
+    #[test]
+    fn test_race_between_destroy_and_trim() {
```

Comment:
I don't see why there is a race

Reply:
Before this PR, race is probably introduced by https://github.com/tikv/tikv/blob/d1d29203e6a93b05dd435ea27a9b39fb30b23f41/components/raftstore-v2/src/operation/ready/mod.rs#L111-L121

After this PR, trim task is moved to background, it can be reordered with any other tasks.